### PR TITLE
Explicitly manage apache2 log rotation

### PR DESCRIPTION
**Pull request by tj90241**
Allow for time and splay of apache2 reload/log rotation
across headnodes to be configured via chef.

### CODE REVIEW

File: chef/cookbooks/bcpc/recipes/apache2.rb [hunk 1]
```diff
@@ -29,6 +29,28 @@
...
 
+# log rotation configuration
+template '/etc/cron.d/logrotate-apache2' do
+  source 'apache2/cron-logrotate.erb'
+  variables(
+    headnodes: headnodes(all: true)
+  )
+end
+
+remote_file '/etc/logrotate-apache2.conf' do
```
Comment (comay) [inline chef/cookbooks/bcpc/recipes/apache2.rb:40]: This looks good although could we put this file under `/etc/apache2` as in something like `/etc/apache2/logrotate.conf`?

File: chef/cookbooks/bcpc/attributes/apache2.rb [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+# the cron.daily tasks.
+#
+# These attributes serve to control both aspects of reloads/log rotations.
+#   * start_hour and start_minute defines the hour and minute, respectfully,
+#     at which to start bounces/log rotations for apache2.
+#   * ...during which process, each headnode's reload will be separated by
+#     splay_minutes.
+default['bcpc']['apache2']['logrotation']['start_hour'] = 22
+default['bcpc']['apache2']['logrotation']['start_minute'] = 0
+default['bcpc']['apache2']['logrotation']['splay_minutes'] = 10
```
Comment (comay) [inline chef/cookbooks/bcpc/attributes/apache2.rb:17]: Do we have a sense on how long the whole Apache logrotation/reload dance takes for a head node? I would have thought a smaller splay would be preferable but if it really takes this long...

File: chef/cookbooks/bcpc/recipes/apache2.rb (hunk 1 above)
Comment (tj90241) [reply chef/cookbooks/bcpc/recipes/apache2.rb:40]: 👍 , will push a change

File: chef/cookbooks/bcpc/attributes/apache2.rb (hunk 2 above)
Comment (tj90241) [reply chef/cookbooks/bcpc/attributes/apache2.rb:17]: Probably takes under a minute to converge; I have no concerns dropping the splay to 1 minute.

Comment (tj90241) [reply chef/cookbooks/bcpc/attributes/apache2.rb]: Dropped to 1 minute.

**Review by comay — APPROVED** [review]
Very nice work, Tyler.

**Pull request merged**